### ISSUE TITLE
Numeric handling in Noto Nastaliq Urdu

### ISSUE THREAD

**Issue by simoncozens** [googlefonts/noto-source#164]
It looks like there ought to be support (through GSUB lookups 9 and 10) for variant Arabic numerals, but these are not applied:

```
$ hb-shape --language='FAR' NotoNastaliqUrdu-Regular.ttf '٤٥٦'
[SixArabic=2+1161|FiveArabic=1+1161|FourArabic=0+1157]
$ hb-shape --language='SND' NotoNastaliqUrdu-Regular.ttf '٤٥٦'
[SixArabic=2+1161|FiveArabic=1+1161|FourArabic=0+1157]
$ hb-shape --language='URD' NotoNastaliqUrdu-Regular.ttf '٤٥٦'
[SixArabic=2+1161|FiveArabic=1+1161|FourArabic=0+1157]
```

The relevant feature code is:

```
# Lookup 9
ccmp { [SixUrdu SevenUrdu] -> [SixArabic SevenFarsi] <arab/SND> };
# Lookup 10
ccmp {
  [FourUrdu SixUrdu SevenUrdu] -> [FourFarsi SixFarsi SevenFarsi] <arab/FAR>;
  DecimalPointArabic -> DecimalPointFarsi <arab/FAR>;
}
```

However, there are no lookups which turn `[FourArabic SixArabic SevenArabic]` into `[FourUrdu SixUrdu SevenUrdu]` so these lookups are never called.

**Comment by anthrotype** [comment]
Hi Simon. Are you saying that the source feature files contain the rules but the compiled OTFs do not?
or that the source feature files are missing the rules altogether?

**Comment by simoncozens** [comment]
Rules to go from Arabic numbers to Urdu numbers are missing altogether from source files and compiled files.

**Comment by punchcutter** [comment]
There are no rules to go from Arabic to Urdu. The default is the Urdu style which are encoded in the range 06F0-06F9 and are the default on the Urdu keyboard. The Arabic are encoded 0660-0669.

**Comment by simoncozens** [comment]
Oh hell, it's one of them. If you want me, I'll be over here saying rude things about Han unification and consistency.

**Comment by marekjez86** [comment]
Thank you @simoncozens

**Comment by simoncozens** [comment]
This one, however, I think *is* a bug:

```
$ hb-view --language='URD' ~/Library/Fonts/NotoNastaliqUrdu-Regular.ttf -u '06dd 0665 06dd 06f5 06dd 0666 06dd 06f6 06dd 0667 06dd 06f7' -O png | imgcat
```
![Screenshot 2020-03-14 at 18 25 38](https://user-images.githubusercontent.com/106728/76688049-35c0c080-6621-11ea-9b63-94b03d2f7f23.png)

**Comment by punchcutter** [comment]
Yes, that does look like a mistake.

**Comment by simoncozens** [comment]
This got fixed in 3.0!
